src/web/rustdoc.rs

```diff
@@ -562,7 +563,7 @@ pub fn badge_handler(req: &mut Request) -> IronResult<Response> {
                 return Ok(super::redirect(iron_url));
             }
 
-            None => BadgeOptions {
+            Err(_) => BadgeOptions {
```

Comment:
Can you add a separate badge for `VersionNotFound` that say "version not found" or something?
```suggestion
            Err(Nope::CrateNotFound) => BadgeOptions {
```